File: untitled/src/main/katas/lupe/codewars/katas/java/maskify/Maskify.java
package lupe.codewars.katas.java.maskify;
public class Maskify {
    public String maskify(String str) {
        return str.length() < 4 ? str
                : str.replaceAll(".", "#")
                .substring(0, str.length() - 4)
                + str.substring(str.length() - 4);
    }
}

File: java-kotlin-katas/src/main/katas/lupe/codewars/katas/kotlin/maskify/Maskify.kt
package lupe.codewars.katas.kotlin.maskify

class Maskify {
    fun maskify(cc: String): String {
        return if (cc.length < 4) cc
                else cc.replace(".".toRegex(), "#")
                        .substring(0, cc.length - 4) +
                        cc.substring(cc.length -4)
    }
}

